service/controller.go

```diff
@@ -155,20 +155,41 @@ const (
 
 var interestingParameters = [...]string{0: "FsType", 1: KeyMkfsFormatOption, 2: KeyBandwidthLimitInKbps, 3: KeyIopsLimit}
 
+type ZoneContent struct {
+	systemID         string
+	protectionDomain ProtectionDomainName
+	pool             PoolName
+}
+
 func (s *service) CreateVolume(
 	ctx context.Context,
 	req *csi.CreateVolumeRequest) (
 	*csi.CreateVolumeResponse, error,
 ) {
 	params := req.GetParameters()
+	var systemID string
+	var err error
 
-	systemID, err := s.getSystemIDFromParameters(params)
-	if err != nil {
-		return nil, err
+	// This is a map of zone to the arrayID and pool identifier
+	zoneTargetMap := make(map[ZoneName]ZoneContent)
+
+	if _, ok := params[KeySystemID]; !ok {
+		zoneTargetMap = s.getZonesFromSecret()
 	}
 
-	if err := s.requireProbe(ctx, systemID); err != nil {
-		return nil, err
+	if len(zoneTargetMap) == 0 {
+		sid, err := s.getSystemIDFromParameters(params)
+		if err != nil {
+			return nil, err
+		}
+
+		systemID = sid
+	}
+
+	if systemID != "" {
+		if err := s.requireProbe(ctx, systemID); err != nil {
```

Comment:
Should we do this probe after we find the systemID based on the zoning information? Add it around line 262 after we found the zoning system id on line 235.

Reply:
Ignore this since I see we perform another probe while finding systemID from the zone info and attempt other probes if it fails.

Reply:
Right, this is for backwards compatibility as this is where it was done if the storage class contains a system ID.